phoenix-core/src/main/java/org/apache/phoenix/schema/PTableImpl.java

```diff
@@ -1559,7 +1586,29 @@ public PName getParentTableName() {
     @Override
     public PName getParentName() {
         // a view on a table will not have a parent name but will have a physical table name (which is the parent)
-        return (type!=PTableType.VIEW || parentName!=null) ? parentName : getPhysicalName();
+        // Update to above comment: we will return logical name of view parent base table
```

Comment:
Why the parent base table and not the immediate parent?

Reply:
This is the existing logic. If this is a child view of another view, the parent is the immediate parent view. If it is a view directly on top of a table, the parent is the logical name of the table. Removed that comment.